algosdk/future/transaction.py

```diff
@@ -2906,6 +2911,83 @@ def __eq__(self, other):
         return False
 
 
+class StateProofTxn(Transaction):
+    """
+    Represents a state proof transaction
+
+    Arguments:
+        sender (str): address of the sender
+        state_proof (dict())
+        state_proof_message (dict(), optional)
+        state_proof_type (str, optional): state proof type
+        sp (SuggestedParams): suggested params from algod
+
+
+    Attributes:
+        sender (str)
+        sprf (dict())
+        sprfmsg (dict())
+        sprf_type (str)
+        first_valid_round (int)
+        last_valid_round (int)
+        genesis_id (str)
+        genesis_hash (str)
+        type (str)
+    """
+
+    def __init__(
+        self,
+        sender,
+        sp,
+        state_proof,
+        state_proof_message=None,
+        state_proof_type=None,
+    ):
+        Transaction.__init__(
+            self, sender, sp, None, None, constants.stateproof_txn, None
+        )
+
+        self.sprf_type = state_proof_type
+        self.sprf = state_proof
+        self.sprfmsg = state_proof_message
+
+    def dictify(self):
+        d = dict()
+        if self.sprf_type:
+            d["sptype"] = self.sprf_type
+        if self.sprfmsg:
+            d["spmsg"] = self.sprfmsg
+        d["sp"] = self.sprf
+        d.update(super(StateProofTxn, self).dictify())
+        od = OrderedDict(sorted(d.items()))
+
+        return od
+
+    @staticmethod
+    def _undictify(d):
+        args = {}
+        if "sptype" in d:
+            args["state_proof_type"] = d["sptype"]
+        if "sp" in d:
+            args["state_proof"] = d["sp"]
+        if "spmsg" in d:
+            args["state_proof_message"] = d["spmsg"]
+
+        return args
+
+    def __eq__(self, other):
+        if not isinstance(other, StateProofTxn):
+            return False
+        return (
+            super(StateProofTxn, self).__eq__(other)
+            and self.sprf_type == other.state_proof_type
```

Comment:
I guess I'm getting it wrong but is other the same object as self? i.e shouldn't that be and self.sprf_type == other.sprf_type

Reply:
you're right. I'm fixing it.